app/src/main/java/com/portfolio/tasky/entry/usecases/domain/UserPreferencesImpl.kt

```diff
@@ -0,0 +1,26 @@
+package com.portfolio.tasky.entry.usecases.domain
```

Comment:
This does neither belong in a use case package nor in domain, it's a data related thing. Probably just a mistake I assume :)

Reply:
okay..wasn't a mistake :D

still need a lot of info to understand this :)

Reply:
One rule of thumb

If it's android related, or it handles data, then it's 100% in the data layer

SharedPreferences is Android related
SharedPreferences ALSO handles data
So always in Data layer

Reply:
Yeah when I learnt about this, I was also often confused because with data we often only associate things like a database or remote API. But, in the end we mean anything "external", so something outside of our app. This can be a remote backend, it can be a local database, it can be saving into a file, but also things like Android specific APIs, such as `NotificationManager`. In that case, it's not your app that shows the notification, instead it's a service from the OS which your app talks to, but still considered external. 

So in your domain layer you shouldn't have any imports that contain android or a 3rd party library.